clients/src/main/java/org/apache/kafka/common/metrics/MetricsReporter.java

```diff
@@ -24,13 +24,14 @@
 import org.apache.kafka.common.Reconfigurable;
 import org.apache.kafka.common.annotation.InterfaceStability;
 import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.telemetry.ClientTelemetry;
 
 /**
  * A plugin interface to allow things to listen as new metrics are created so they can be reported.
  * <p>
  * Implement {@link org.apache.kafka.common.ClusterResourceListener} to receive cluster metadata once it's available. Please see the class documentation for ClusterResourceListener for more information.
  */
-public interface MetricsReporter extends Reconfigurable, AutoCloseable {
+public interface MetricsReporter extends Reconfigurable, AutoCloseable, ClientTelemetry {
```

Comment:
I think this is incorrect. MetricsReporter does not extend ClientTelemetry. Actually, an instance of MetricsReporter might also implement ClientTelemetry. With this scheme, every MetricsReporter would also implement ClientTelemetry.

Reply:
It does seem like it should be the reverse. But given that the `ClientTelemetry` interface's methods both have defaults, this wouldn't technically break anything, would it? Still, I don't think this tight coupling is desired.

Reply:
I was in an impression that we want Future MetricsReporter to support either client or server side mapping. And that's why this change existed in v1 of the code as well.

Looking at it again seems I am creating `fat interfaces` and doesn't make much sense to create common `ClientTelemetry` interface with optional client and server behaviour rather split them. And let the implementation of MetricsReporter decide which client or server side additional provider it wants to support. I ll change it.

Reply:
I think the relationship between client/server and public/internal interfaces was unclear to me. It just needs a bit of tidying up.

Reply:
I have removed MetricsReporter changes itself, resolving the comment.

Reply:
the MetricsReporter plugin is a plugin for either server or client side. However the ClientTelemetry plugin is only relevant for broker side plugins

Reply:
I agree with Xavier. Also, the KIP didn't propose to change the `MetricsReporter` interface.

Reply:
@xvrl @junrao I agree as well and checked in code do not have the change in MetricsReporter.java interface.